hips/hip-9999.md

```diff
@@ -0,0 +1,220 @@
+
+Chart authors should also use the `kubeVersion` field to specify compatible Kubernetes versions, but this may not be enough to avoid breaking changes based on other resources and controllers is the cluster.
+
+## Rationale
+
+If helm is "the package manager for Kubernetes" it should have similar package manager features that users are used to such as `apt` and `yum`.
+With both of those package managers there is a centralized way to specify a version of a package and keep that version installed even when a user runs an `upgrade` command against a release.
+Centralizing the desired version guarantees that all users, automation, and libraries that use the package manager also honor the version locks in place.
+
+Making something native in helm that supports locking will be critical as users consume helm charts via other tools and workflows (e.g. gitops via `flux`).
+It will reduce errors because `helm` commands can be run from any users machine with access to the cluster.
+This is opposed to a helm plugin model where a user or tool may not have the plugin installed.
+
+Helm releases do not have the same constraints has traditional package managers.
+A cluster can have multiple versions of a chart deployed in different namespaces and each release can have various dependencies.
+Because releases are self-contained on a per-release basis, version locking information should also exist within the release metadata and be read and processed during template rendering.
+
+## Specification
+
+To implement version locking on a release an optional field should be added to the release secret that specifies a desired version.
+
+```
+apiVersion: v1                                                           
+data:                                                                 
+  release: SDRzSUF...
+kind: Secret
+metadata:
+  creationTimestamp: "2021-02-23T18:07:25Z"
+  desiredChartVersion: "8.0.*"             <---- this field is added
+  labels:
+    modifiedAt: "1614103661"
+    name: ingress
+    owner: helm
+    status: deployed
+    version: "1"
+  name: sh.helm.release.v1.ingress.v1
+  namespace: default
+  resourceVersion: "47331598"
+  selfLink: /api/v1/namespaces/default/secrets/sh.helm.release.v1.ingress.v1
+  uid: 07ba4723-1cd1-43a8-a322-7d8e25f0868e
+type: helm.sh/release.v1
+```
+_Field name and exact json path may change_
+
+Having a desired chart version will allow customers to stay on specific chart versions based on the published chart version data.
+The chart field is required to be SemVer 2 version and the `desiredChartVersion` can be an exact version match (eg `1.0.0`) or a wildcard string match (e.g. `1.*`).
+
+The field can be added with the `helm lock` command.
+It will take a release name and chart version to lock to.
+```
+helm lock autoscaler --version '8.*'
+```
```

Comment:
Why does `lock` to `8.*` create a lock to `8.0.*` (example above)?

Reply:
I didn't mean that `8.*` would become `8.0.*` I was simply providing different examples of how a user could lock their chart to a major or minor version of the chart.